designsystem/src/commonMain/kotlin/org/hisp/dhis/mobileui/designsystem/component/RadioButton.kt

```diff
@@ -0,0 +1,102 @@
+import org.hisp.dhis.mobileui.designsystem.theme.DHISTypography
+import org.hisp.dhis.mobileui.designsystem.theme.Outline
+import org.hisp.dhis.mobileui.designsystem.theme.Spacing
+import org.hisp.dhis.mobileui.designsystem.theme.SurfaceColor
+import org.hisp.dhis.mobileui.designsystem.theme.TextColor
+
+/**
+ * DHIS2 radio button with or without text. Wraps Material 3 [RadioButton].
+ *
+ * @param selected Controls the selected option state for multiple options.
+ * @param enabled Controls the enabled state of the button. When `false`, this button will not be
+ * clickable and will appear disabled to accessibility services.
+ * @param textInput The button option text.
+ * @param onClick Will be called when the user clicks the button.
+ */
+@Composable
+fun RadioButton(
+    selected: Boolean,
+    enabled: Boolean,
+    textInput: String?,
+    onClick: () -> Unit
+) {
+    CompositionLocalProvider(LocalRippleTheme provides CustomDHISRippleTheme) {
+        if (textInput.isNullOrEmpty()) {
+            RadioButton(
+                selected = selected,
+                onClick = onClick,
+                enabled = enabled,
+                modifier = Modifier
+                    .size(Spacing.Spacing40)
+                    .padding(Spacing.Spacing8),
+                colors = RadioButtonDefaults.colors(
+                    selectedColor = SurfaceColor.Primary,
+                    unselectedColor = Outline.Dark,
+                    disabledSelectedColor = TextColor.OnDisabledSurface,
+                    disabledUnselectedColor = TextColor.OnDisabledSurface
+                )
+            )
+        } else {
+            Row(
+                modifier = Modifier.width(840.dp).height(Spacing.Spacing40),
+                horizontalArrangement = Arrangement.spacedBy(0.dp, Alignment.Start),
+                verticalAlignment = Alignment.Top
+            ) {
+                RadioButton(
+                    selected = selected,
+                    onClick = onClick,
+                    enabled = enabled,
+                    modifier = Modifier
+                        .size(Spacing.Spacing40),
+                    colors = RadioButtonDefaults.colors(
+                        selectedColor = SurfaceColor.Primary,
+                        unselectedColor = Outline.Dark,
+                        disabledSelectedColor = TextColor.OnDisabledSurface,
+                        disabledUnselectedColor = TextColor.OnDisabledSurface
+                    )
+                )
+                Row(
+                    modifier = Modifier
+                        .width(800.dp)
+                        .height(40.dp)
+                        .padding(top = Spacing.Spacing8, bottom = Spacing.Spacing8),
+                    horizontalArrangement = Arrangement.spacedBy(0.dp, Alignment.Start),
+                    verticalAlignment = Alignment.Top
+                ) {
+                    Text(
```

Comment:
I think that this can be simplified, instead of the if else, where you are writing duplicated code (RadioButton), maybe you can just use the else part with the row, radiobutton and text. Then, you can check if text.isNullOrEmpty() to draw the text component:
`
if (!textInput.isNullOrEmpty()) {
Text(...)
}
`
What do you think?